src/senselab/audio/tasks/speaker_verification/speaker_verification.py

```diff
@@ -0,0 +1,59 @@
+"""Audio Processing and Speaker Verification Module.
+
+This module provides functions for resampling audio using an IIR filter and
+verifying if two audio samples or files are from the same speaker using a
+specified model.
+"""
+
+from typing import List, Tuple
+
+from torch.nn.functional import cosine_similarity
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.audio.tasks.speaker_embeddings.speechbrain import SpeechBrainEmbeddings
+from senselab.utils.data_structures.device import DeviceType
+from senselab.utils.data_structures.model import SpeechBrainModel
+
+
+def verify_speaker(
+    audios: List[Tuple[Audio, Audio]],
+    model: SpeechBrainModel = SpeechBrainModel(path_or_uri="speechbrain/spkrec-ecapa-voxceleb", revision="main"),
+    model_training_sample_rate: int = 16000,  # spkrec-ecapa-voxceleb trained on 16kHz audio
+    device: DeviceType = DeviceType.CPU,
```

Comment:
can you please make device's default always as None and use our _select_device_and_dtype method? this way, if the user doesn't have any preference and a GPU is available, it will be used. in your code, you force the user to use a CPU no matter what

Reply:
With MPS, was getting:
```
ValueError: The requested DeviceType is either not available or                             compatible with this functionality.

src/senselab/utils/data_structures/device.py:60: ValueError
```
I excluded it from compatible_devices.

Reply:
yup. speechbrain models don't support mps yet. excluding it from compatible_devices is the way to go. idk why i cannot see your last changes though and keep seeing `device: DeviceType = DeviceType.CPU`. am i missing something?

Reply:
Can you see it now?